core/src/main/java/com/graphhopper/GraphHopper.java

```diff
@@ -1093,10 +1096,15 @@ public GHResponse route( GHRequest request )
         QueryResult fromQResult = qResults.get(0);
 
         double weightLimit = request.getHints().getDouble("defaultWeightLimit", defaultWeightLimit);
+        int maxVisistedNodesForRequest = request.getHints().getInt("routing.max_visited_nodes", maxVisitedNodes);
+        // TODO Do we want to throw an exception at this point?
```

Comment:
yes, I would prefer throwing an exception or even better: adding an error to the response